src/vs/workbench/parts/search/common/search.ts

```diff
@@ -53,16 +69,20 @@ export namespace WorkspaceSymbolProviderRegistry {
 		};
 	}
 
-	export function all(): IWorkspaceSymbolProvider[] {
-		return _supports.slice(0);
+	export function all(workspace?: IWorkspace): IWorkspaceSymbolProvider[] {
+		const supports = _supports.slice(0);
```

Comment:
This is only to create a copy right? The filter case already creates a copy so you could do this

```
if (workspace) {
    return _supports.filter(support => support.workspace && support.workspace.resource && support.workspace.resource.toString() === workspace.resource.toString())
}
return _supports.slice(0); // create a copy
```